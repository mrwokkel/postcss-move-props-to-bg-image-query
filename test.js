const postcss = require('postcss');

const plugin = require('.');

function exec(input, opts) {
  return postcss()
    .use(plugin(opts))
    .process(input, { from: __filename })
    .then(({ css }) => css);
}

it('should move -svg-* props by default', async () => {
  expect(await exec(
    '.a {background: url(1.png); fill: red}'
  )).toMatchSnapshot();

  expect(await exec(
    '.a {background: url(1.png); -svg-fill: red}'
  )).toMatchSnapshot();
});

it('should preserve if any query params already there', async () => {
  expect(await exec(
    '.a {background: url(1.png?qwe); -svg-fill: red}'
  )).toMatchSnapshot();
});

it('should allow to override default props', async () => {
  expect(await exec(
    '.a {background: url(1.png); -svg-fill: red; color: blue}',
    { match: ['color'] }
  )).toMatchSnapshot();
});

it('should allow to use custom transformer', async () => {
  expect(await exec(
    '.a {background: url(1.png); -svg-fill: red;}',
    { transform: ({ name, value }) => ({ name: name.toUpperCase(), value }) }
  )).toMatchSnapshot();
});

it('should always encode prop value even with custom transformer', async () => {
  expect(await exec(
    '.a {background: url(1.png); -svg-fill: #fff;}'
  )).toMatchSnapshot();

  expect(await exec(
    '.a {background: url(1.png); -svg-fill: #fff;}',
    { transform: ({ name, value }) => ({ name: name.toUpperCase(), value }) }
  )).toMatchSnapshot();
});
