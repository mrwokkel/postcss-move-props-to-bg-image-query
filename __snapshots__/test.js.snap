// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should allow to override default props 1`] = `".a {background: url(1.png?color=blue); -svg-fill: red}"`;

exports[`should allow to use custom transformer 1`] = `".a {background: url(1.png?-SVG-FILL=red);}"`;

exports[`should always encode prop value even with custom transformer 1`] = `".a {background: url(1.png?fill=%2523fff);}"`;

exports[`should always encode prop value even with custom transformer 2`] = `".a {background: url(1.png?-SVG-FILL=%2523fff);}"`;

exports[`should move -svg-* props by default 1`] = `".a {background: url(1.png); fill: red}"`;

exports[`should move -svg-* props by default 2`] = `".a {background: url(1.png?fill=red)}"`;

exports[`should preserve if any query params already there 1`] = `".a {background: url(1.png?qwe&fill=red)}"`;
